### ISSUE TITLE
Duplicate node added

### ISSUE THREAD

**Issue by baaiy0610** [spcl/dace#1050]
**Describe the bug**
I plan to write a function to find the tangent vector on the sphere along the longtitude.
first i need transfer the 3d point into latlng coordinate system and after transforming i need transfer the latlng point back to 3d point. 
When i use dace.program with latlng_to_xyz it shows Duplicate node added, but it works when i delete it.
 
The issue appears when calling `latlng_to_xyz()`in the function `vec_sphere_lng()` 
**To Reproduce**
1. run latlng_to_xyz with @dace.program, it shows error
2. run latlng_to_xyz without @dace.program, it can return value as normal (array([ 0.11697778, -0.3213938 ,  0.93969262]))

**Expected behavior**
As i dont't use the dace.program 

**Screenshots**
![image](https://user-images.githubusercontent.com/53456502/178331023-56eef678-1a55-4d5c-be78-b6b97c500e1b.png)
```python
@dace.program
def xyz_to_latlng(vert_3d:dace.float64[3], r:dace.float64)->dace.float64[2]:
    vert_sphere = np.zeros((2,))
    
    vert_sphere[0] = np.rad2deg(np.arcsin(vert_3d[2]/r))
    vert_sphere[1] = np.rad2deg(np.arctan2(vert_3d[1], vert_3d[0]))
    return vert_sphere

@dace.program
def latlng_to_xyz(vert_sphere:dace.float64[2], r:dace.float64)->dace.float64[3]:
    vert_3d = np.zeros((3,))
        
    vert_3d[0] = r * np.cos(vert_sphere[0]*np.pi/180.) * np.cos(vert_sphere[1]*np.pi/180.)
    vert_3d[1] = r * np.cos(vert_sphere[0]*np.pi/180.) * np.sin(vert_sphere[1]*np.pi/180.)
    vert_3d[2] = r * np.sin(vert_sphere[0]*np.pi/180.)
    return vert_3d

@dace.program
def vec_sphere_lng(point1: dace.float64[3], point2:dace.float64[3], r:dace.float64)->dace.float64[3]:
    
    mid_point = (point1 + point2)/2.0
    mid_vector = mid_point * r / np.sqrt(mid_point[0]**2 + mid_point[1]**2 + mid_point[2]**2)
    mid_vector_sphere = xyz_to_latlng(mid_vector, r)
    
    if mid_vector_sphere[0] > 0.0:
        mid_vector_sphere[0] -= 90.0
        vector_sphere = latlng_to_xyz(mid_vector_sphere, r)
        return -vector_sphere
    
    elif mid_vector_sphere[0] < 0.0:
        mid_vector_sphere[0] += 90.0
        vector_sphere = latlng_to_xyz(mid_vector_sphere, r)
        return vector_sphere
    
    elif mid_vector_sphere[0] == 0.0:
        return np.array([0., 0., 1.])

a = np.array([10.,110.])
b = np.array([30.,110.])
r = 1.0
a_xyz = latlng_to_xyz(a,r)
b_xyz = latlng_to_xyz(b,r)

vec_sphere_lng(a_xyz, b_xyz, r)

**Comment by tbennun** [comment]
@baaiy0610 We managed to reproduce the issue and it has to do with the fact that you call `latlng_to_xyz` before calling `vec_sphere_lng`. While we are fixing the issue, the simplest way to work around it is to comment out the `@dace.program` from both small functions. 

DaCe will in any case parse and compile them (it does that for every called function). In general it is convenient to only annotated the top-level function that you will call.

I will keep the issue open until it is fixed.

**Comment by tbennun** [comment]
The issue is fixed in #1056. It will be closed automatically upon merge.
